### PR TITLE
[WIP] fix(modal): fix modal footer height

### LINKED ISSUE #1000: [Modal]: Too much content or a very short window causes the buttons to be pushed off the footer
## Detailed description

When using a transactional modal it is possible to push the buttons off of the footer. This happens when you either have too much content in the modal or if you have a short window.

> Is this issue related to a specific component? 

Modal component

> What did you expect to happen? What happened instead? What would you like to see changed?

If adding more content it shouldn't affect the placement of the buttons since the content has scroll.

> What browser are you working in? 

Chrome, tested in Firefox and the same does not happen

> What version of the Carbon Design System are you using? 

Last tested in 9.5.0

## Steps to reproduce the issue

1. Open up the [react storybook modal component](http://react.carbondesignsystem.com/?selectedKind=ModalWrapper&selectedStory=transactional%20modal&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) in Chrome
2. Open the transactional modal
3. Shrink the window vertically until it is very small and watch the buttons be pushed off the footer
4. Make the window a normal size, inspect the modal content and copy and paste the text of one of the content paragraphs over and over inside dev tools and watch as the button moves off the footer.

## Additional information

image

image

### PR DESCRIPTION
**Pull request by aledavila**
Closes #1000 

Fixes button repositioning in footer. 

#### Changelog

**Changed**

- removed `display: flex` and added `inline-block` to footer and `text-align: right` to parent for flex-box positioning as flex was breaking the position of buttons. 

#### Testing / Reviewing

Check both react and vanilla modal and transactional modal.

### CODE REVIEW

**Comment by asudoh** [comment]
Hi @aledavila thank you for jumping in! What I have got in mind is custom modal footers - The change drifting away from `display:flex` would impact that. Given this, we may consider the change in a future major release, have applications impacted by the issue apply the proposed style by themselves?

**Comment by aledavila** [comment]
Hey @asudoh thanks for jumping in. Yea I'm ok with proposing applications to implement the style by themselves.

**Comment by shinytoyrobots** [comment]
Will this actually cause any break? @asudoh - do you have an example of what might be impacted? We don't have custom modal footers, and I'm not sure we should support edge cases where someone else is building custom functionality outside of Carbon.

**Comment by asudoh** [comment]
Hi @shinytoyrobots thank you for asking! e.g. IBM Cloud product tour dialog has a small content switcher at the dialog footer to switch the carousel in the dialog. Given the bottom left area of dialog is a blank space, applications tend to put their stuff in. I understand support of that use case is not declared anywhere, but given how common such use case is, I'd be reluctant to make the change all the sudden.

**Comment by alisonjoseph** [comment]
Should we just somehow track this for `v10` and make the change then?

**Comment by asudoh** [comment]
@alisonjoseph Agreed!

**Comment by alisonjoseph** [comment]
Closing for now then and adding a v10 label to the original issue.

**Comment by aledavila** [comment]
Re-opening for v10 release.

File: src/components/modal/_modal.scss [hunk 1]
```diff
@@ -58,6 +58,7 @@
     max-height: 100%;
     height: 100%;
     padding: $spacing-xl 3% 0rem 3%;
+    text-align: right;
```
Comment (lovemecomputer) [inline src/components/modal/_modal.scss:62]: would it work to move this rule onto a different div?

**Review by lovemecomputer — CHANGES_REQUESTED** [review]
`text-align: right` is applied to `bx--modal-container`, so it makes all of the modal content align right:

screen shot 2018-11-21 at 11 19 10 am

**Comment by emyarod** [comment]
superseded by #1499

**Pull request closed without merging**